tools/filterdiff.py

```diff
@@ -0,0 +1,179 @@
+            # and also a list of unwanted tmpfiles_d to filter out on comparison
+            # time.
+            return cls(c["image-info"], set(c.get("unwanted_tmpfiles_d", [])))
+        # In the case of a raw image-info, just init it with an empty
+        # unwanted_tmpfiles_d set.
+        return cls(c, set())
+
+    def diff(self, other, unwanted_tmpfiles_d: Set, verbose=False) -> bool:
+        """
+        Do perform a diff between this object and an other of the same type.
+
+
+        parameters:
+        -----------
+        other:                  the other ImageInfo object to diff against
+        unwanted_tmpfiles_d:    A list of tmpfiles_d to ignore
+                                All the unwanted_tmpfiles_d found in the
+                                ImageInfo to compare are unioned with the
+                                unwanted_tmpfiles_d given as parameters.
+        verbose:                True to get the diff command output on stdout
+        """
+        unwanted_tmpfiles_d = self.unwanted_tmpfiles_d.union(
+            unwanted_tmpfiles_d,
+            other.unwanted_tmpfiles_d)
+        s = self.imi
+        o = other.imi
+        if self.do_filter:
+            s = FilterImageInfo(self.imi, unwanted_tmpfiles_d).apply()
+        if other.do_filter:
+            o = FilterImageInfo(other.imi, unwanted_tmpfiles_d).apply()
+        if ImageInfo.exec_diff(s, o, verbose):
+            return True
+        return False
+
+    @staticmethod
+    def exec_diff(imi1: Dict, imi2: Dict, visual: bool) -> int:
+        """
+        Write up the two dictionaries to temporary files and invoke `diff` as a
+        subprocess command. Visual set to True to get the result of the diff
+        command on stdout.
+        """
+        with tempfile.TemporaryDirectory() as d:
+            f1 = os.path.join(d, "item1")
+            f2 = os.path.join(d, "item2")
+            try:
+                with open(f1, 'w') as ff1:
+                    json.dump(imi1, ff1, indent=4)
+
+                with open(f2, 'w') as ff2:
+                    json.dump(imi2, ff2, indent=4)
+
+                p = subprocess.run(["diff", f1, f2],
+                                   check=False,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE)
+                if visual:
+                    out = p.stdout.decode("utf-8")
+                    for l in out.split("\n"):
+                        print(l)
+                return p.returncode
+            finally:
+                os.remove(f1)
+                os.remove(f2)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="osbuild image tests")
+    parser.add_argument("files",
+                        nargs=2,
+                        help="files to diff, 2 required")
+    parser.add_argument("--unwanted-tmpfiles-d",
+                        nargs="+",
+                        default=[],
```

Comment:
nitpick: you could have defined this option as:

```python
parser.add_argument("--unwanted-tmpfiles-d",
    nargs="+",
    default=set(),
    type=set,
    help="List of unwanted files to get from tmpfiles_d")
```

By specifying the `type`, [parser can automatically convert the user-provided value](https://docs.python.org/3/library/argparse.html#type).

Reply:
Hmmm, I can't get that to work :thinking: 
While printing what's in `args.unwanted-tmpfiles-d ` from the command `python tools/filterdiff.py manifest-db/centos_9-s390x-qcow2-boot.json manifest-db/centos_9-s390x-qcow2_customize-boot.json --unwanted-tmpfiles-d toto --unwanted-tmpfiles-d titi`
I get:
`[{'i', 't'}]`
Which is not what I want ^^

Reply:
I see 🤔  This is the result of using `type=set`, which I didn't realized when suggesting. What the code does is basically `set(<string>)` which ends up being what you see... You could implement your own handling, but in this case it is probably for the best to just do what you did originally.